File: atom-change-event/src/main/java/nl/haploid/octowight/AtomGroup.java
package nl.haploid.octowight;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;

public class AtomGroup {

	private String atomOrigin;

	private String atomType;

	public String getAtomOrigin() {
		return atomOrigin;
	}

	public void setAtomOrigin(final String atomOrigin) {
		this.atomOrigin = atomOrigin;
	}

	public String getAtomType() {
		return atomType;
	}

	public void setAtomType(final String atomType) {
		this.atomType = atomType;
	}

	@Override
	public boolean equals(final Object that) {
		return EqualsBuilder.reflectionEquals(this, that, false);
	}

	@Override
	public int hashCode() {
		return HashCodeBuilder.reflectionHashCode(this, false);
	}
}

File: atom-change-event/src/main/scala/nl/haploid/octowight/AtomGroup.scala
package nl.haploid.octowight

import org.apache.commons.lang3.builder.{EqualsBuilder, HashCodeBuilder, ToStringBuilder}

import scala.beans.BeanProperty

object AtomGroup {

  def apply(origin: String, category: String): AtomGroup = {
    val g = new AtomGroup
    g.setCategory(category)
    g.setOrigin(origin)
    g
  }
}

class AtomGroup {
  @BeanProperty var origin: String = _
  @BeanProperty var category: String = _

  override def equals(that: Any) = EqualsBuilder.reflectionEquals(this, that, false)

  override def hashCode = HashCodeBuilder.reflectionHashCode(this, false)

  override def toString = ToStringBuilder.reflectionToString(this)
}
